src/io/obf.rs

```diff
@@ -301,9 +310,61 @@ where
     );
     log_mem("Sampled final_preimage");
 
+    #[cfg(feature = "store")]
+    {
+        let timestamp_id = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .expect("Time went backwards")
+            .as_nanos() as u64;
+
+        let dir_path = PathBuf::from(format!("obf_{timestamp_id}"));
+
+        if !dir_path.exists() {
+            std::fs::create_dir_all(&dir_path).expect("Failed to create directory");
+        }
+
+        let start = SystemTime::now();
```

Comment:
Can you use `std::time::Instant` to measure time in the same way as e2e io tests?
(I don't which is better, but I just want to ensure that the same method is used to measure time.)